src/main/java/org/terasology/accelerationblocks/component/AccelerationComponent.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.accelerationblocks.component;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.accelerationblocks.AccelerationSystem;
+import org.terasology.entitySystem.Component;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3f;
+
+public final class AccelerationComponent implements Component {
+    public Vector3f velocity;
+
+    public boolean ignoreBlockDirection;
+
+    public AccelerationComponent() {
+    }
+
+    public void initialize(Vector3f velocity, boolean ignoreBlockDirection){
```

Comment:
Generally (if not always) we try to avoid all logic in Components including constructors. Fields are just public and set directly by whatever is creating the Component.

(This may be inconsistently applied throughout the code base, I admit)